orodha-keycloak/orodha_keycloak/connection.py

```diff
@@ -2,14 +2,77 @@
 This Module contains the OrodhaKeycloakClient class which is a facade
 used to interact with a keycloak server via python-keycloak.
 """
+import os
 import orodha_keycloak.connections.admin
 import orodha_keycloak.connections.client
-from orodha_keycloak.connections.exceptions import InvalidConnectionException
+from orodha_keycloak.exceptions import InvalidConnectionException
+
+
+class OrodhaCredentials:
+    """
+    A class used to obtain and package credentials for the Orodha system.
+    If kwargs are not passed in, The credentials are pulled from the environment.
+
+    Args(as kwargs):
+        server_url(str): The url of the server that our keycloak is hosted at
+        realm_name(str): The name of the keycloak realm that we are attempting to access.
+        client_id(str): The keycloak client_id that we are using for the connection.
+        client_secret_key(str): The secret key of the keycloak client.
+        username(str) - Optional: The username of the user being impersonated by python-keycloak
+        password(str) - Optional: The password of the user being impersonated by python-keycloak
+
+    Raises:
+        InvalidConnectionException: If required arguments are not available
+            as kwargs or in the environment.
+    """
+
+    def __init__(self, **kwargs):
+        arg_dict = {
+            "server_url": kwargs.get("server_url"),
+            "realm_name": kwargs.get("realm_name"),
+            "client_id": kwargs.get("client_id"),
+            "client_secret_key": kwargs.get("client_secret_key"),
+            "username": kwargs.get("username"),
```

Comment:
@Edison-Stuart I missed it before, but you'll want to do two things here:

1. Use upper case names for the env vars
2. Use more specific names for the env vars

e.g. instead of `client_secret_key` in the env, you'd want something like 'ORODHA_KEYCLOAK_CLIENT_SECRET`

Reply:
Okay, I changed the names to be more descriptive as well as made it so all the environment variables are capitalized.